### DISCUSSION TITLE
What does .gitignore do?

### DISCUSSION THREAD

**Discussion by Jme7** [community/community#22729] [New to GitHub]
<p>I know, this is a newbie question… and this may be the wrong category to put this in. But I don’t understand this. Can anyone help?</p>
<p>Thanks in advance! :heart:</p>

**Comment by SethClydesdale** [comment, marked as answer]
<p>Hello <a class="mention" href="/jme7">@Jme7</a>,</p>
<p><code>.gitignore</code> is used for excluding certain files or directories from your commits that you don’t want committed to the main repository, such as temporary files created by IDEs for example. More info <a href="https://git-scm.com/docs/gitignore" rel="nofollow noopener">here</a>.</p>
<p>As another, more simple example, on one of my projects I keep a folder called “OFFLINE_ASSETS” where I have binaries, personal notes, and some pre-production stuff that I don’t want to clutter the repo with. The contents look like this:</p>
<pre><code class="lang-auto">.gitignore
OFFLINE_ASSETS/
</code></pre>
<p>When committing changes, both the .gitignore file and OFFLINE_ASSETS folder are ignored.</p>

**Comment by Jme7** [comment]
<p><a class="mention" href="/SethClydesdale">@SethClydesdale</a> Ok, now I understand that. Thanks!</p>
